fineract-doc/src/docs/en/chapters/features/buydown-fee.adoc

```diff
@@ -0,0 +1,637 @@
+  "buyDownFeeCalculationType": "FLAT",
+  "buyDownFeeStrategy": "EQUAL_AMORTIZATION", 
+  "buyDownFeeIncomeType": "FEE",
+  "buyDownExpenseAccountId": 123,
+  "incomeFromBuyDownAccountId": 456,
+  // ... other loan product parameters
+}
+----
+
+==== Disabling Buy Down Fee
+
+[source,json]
+----
+{
+  "enableBuyDownFee": false
+  // buyDownFee* parameters are optional when disabled
+}
+----
+
+=== Error Scenarios
+
+Common validation errors and their resolutions:
+
+* **"Buy Down Fee calculation type is required"**: Provide `buyDownFeeCalculationType` when enabling Buy Down Fee
+* **"Buy Down Fee strategy is required"**: Provide `buyDownFeeStrategy` when enabling Buy Down Fee  
+* **"Buy Down Fee income type is required"**: Provide `buyDownFeeIncomeType` when enabling Buy Down Fee
+* **"Buy Down expense account is required"**: Provide valid `buyDownExpenseAccountId`
+* **"Income from Buy Down account is required"**: Provide valid `incomeFromBuyDownAccountId`
+* **"Buy Down fees can only be added to active loans"**: Ensure loan status is ACTIVE before adding Buy Down Fee transactions
+
+== API Endpoints
+
+=== Create Buy Down Fee Transaction
+
+**Endpoint**: `POST /loans/{loanId}/transactions?command=buyDownFee`
+
+**Alternative**: `POST /loans/external-id/{loanExternalId}/transactions?command=buyDownFee`
+
+**Purpose**: Creates a new Buy Down Fee transaction for a loan, establishing the upfront merchant fee that will be amortized over the loan term.
+
+**Request Parameters**:
+[cols="1,1,1,3"]
+|===
+|Parameter |Type |Required |Description
+
+|`transactionDate`
+|String
+|Yes
+|Transaction date in specified format (e.g., "dd MMMM yyyy")
+
+|`dateFormat`
+|String
+|Yes
+|Date format specification (e.g., "dd MMMM yyyy")
+
+|`locale`
+|String
+|Yes
+|Locale for date formatting (e.g., "en")
+
+|`transactionAmount`
+|BigDecimal
+|Yes
+|Buy down fee amount (must be positive)
+
+|`paymentTypeId`
+|Long
+|No
+|Payment type identifier
+
+|`note`
+|String
+|No
+|Transaction note/description
+
+|`externalId`
+|String
+|No
+|External transaction identifier
+|===
+
+**Request Example**:
+[source,json]
+----
+{
+  "transactionDate": "01 January 2024",
+  "dateFormat": "dd MMMM yyyy",
+  "locale": "en",
+  "transactionAmount": 50.0,
+  "note": "Merchant fee for 0% interest loan"
+}
+----
+
+**Response Structure**:
+[source,json]
+----
+{
+  "loanId": 123,
+  "resourceId": 456,
+  "resourceExternalId": "ext-123",
+  "clientId": 789,
+  "officeId": 1
+}
+----
+
+**Business Logic**:
+
+1. Validates that Buy Down Fee is enabled for the loan product
+2. Ensures transaction date is not before first disbursement date
+3. Verifies client/group is active
+4. Creates Buy Down Fee transaction with specified amount
+5. Establishes LoanBuyDownFeeBalance record for amortization tracking
+6. Posts journal entries (Debit: Buy Down Expense, Credit: Deferred Capitalized Income)
+7. Triggers business events for transaction creation and loan balance change
```

Comment:
Loan balance does not change with buy down fees, LoanBalanceChangedBusinessEvent is not fired here